### ISSUE TITLE
Fix maximum call stack size exceeded error

### ISSUE THREAD

**Issue by Dangoo** [stylelint/stylelint#4697]
> Clearly describe the bug

v13.3.1 causes Maximum call stack size exceeded error
The same setup worked smoothly with v13.3.0.
Might be related to [change the parser](https://github.com/stylelint/stylelint/issues/4574#issuecomment-611468118) in a patch release?

> Which rule, if any, is the bug related to?

Hard to make up…

> What code is needed to reproduce the bug?

TBD

> What stylelint configuration is needed to reproduce the bug?

```json
{
  "extends": [
    "stylelint-config-recommended"
  ]
}
```

> Which version of stylelint are you using?

`13.3.1`

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

CLI with `stylelint "**/*.{ts, tsx}"`

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

Occurred (not otherwise tested yet) with Typescript with CSS-in-JS

> What did you expect to happen?

Running smoothly as with v13.3.0

> What actually happened (e.g. what warnings or errors did you get)?

```shell
RangeError: Maximum call stack size exceeded
    at LocalFixer.parse (PROJECT/node_modules/@stylelint/postcss-css-in-js/template-parser-helper.js:197:10)
    at parseStyle (PROJECT/node_modules/@stylelint/postcss-css-in-js/template-parser-helper.js:228:54)
    at PROJECT/node_modules/@stylelint/postcss-css-in-js/template-parser-helper.js:97:17
    at Array.forEach (<anonymous>)
    at parseTemplateLiteralStyles (PROJECT/node_modules/@stylelint/postcss-css-in-js/template-parser-helper.js:96:4)
    at TemplateParser.literal (PROJECT/node_modules/@stylelint/postcss-css-in-js/template-parser-helper.js:55:18)
    at TemplateParser.other (PROJECT/node_modules/@stylelint/postcss-css-in-js/template-parser.js:14:25)
    at TemplateParser.parse (PROJECT/node_modules/postcss/lib/parser.js:77:16)
    at LocalFixer.templateParse (PROJECT/node_modules/@stylelint/postcss-css-in-js/template-parse.js:14:9)
    at LocalFixer.parse (PROJECT/node_modules/@stylelint/postcss-css-in-js/template-parser-helper.js:182:16)
```

**Comment by daviddelusenet** [comment]
I'm experiencing the same error.

**Comment by ota-meshi** [comment]
Could you share a reproducible source code?

**Comment by jeddy3** [comment]
@Dangoo and @daviddelusenet thank you for the reports. This may be a difficult one to track down and we would appreciate your help.

As @ota-meshi requested, do you have reproducible source code to share? 

I've a few more questions that may help us pinpoint where the issue lies.

1 - Are either of you linting source files with nested template literals, e.g.:

```
import styled, { css } from 'styled-components';
const Message = styled.p`
	padding: 10px;
	${(props) => css`
		color: #b02d00;
	`}
`;
```

If so, does stylelint still error when you lint just this file?

2 - How many source files are you linting? You can use the `verbose` formatter to find this out, like so:

```
stylelint "**/*.{ts, tsx}" -f verbose
```

3 - Does stylelint still error when you turn off the `indentation` rule?:

```json
{
  "rules": { 
   ..
  "indentation": null
  }
}
```

These questions touch on areas which were modified in the last release.

**Comment by Dangoo** [comment]
Hi @jeddy3 @ota-meshi, 
thanks for your quick responses!
Sadly I'll not be able to provide a test case over the easter days. But I can answer @jeddy3 s` questions briefly:

> 1 - Are either of you linting source files with nested template literals?

Yes we do

> 2 - How many source files are you linting

`251 sources checked`

> 3 - Does stylelint still error when you turn off the indentation rule?

It still fails with the same error message

Will reach back to you on tuesday.

**Comment by daviddelusenet** [comment]
I'm also nesting template literals so I think there lies the problem.

**Comment by daviddelusenet** [comment]
I'm also nesting template literals so I think that is related to this bug.

**Comment by jeddy3** [comment]
Thank you both for getting back to us. We have passing tests for simple nested template literals, e.g.: 

```js
import styled, { css } from 'styled-components';
const Message = styled.p`
	padding: 10px;
	${(props) => css`
		color: #b02d00;
	`}
`;
```

So there must a particular nested literal pattern we're overlooking. When you have time, would you mind trying to find (by linting your nested template literals files one-by-one) a snippet of source code that triggers the error?

Likewise, if anyone else runs into this problem then doing the same would be very helpful.

> Sadly I'll not be able to provide a test case over the easter days

Of course. I hope you have a good Easter!

---

>  Interestingly `{ts, tsx}` only linted the 88 `.ts` files and ignored the 52 remaining `.tsx` files. But that seems like another issue…

The space after the comma may be causing this. See [this glob test](https://globster.xyz/?q=**%2F*.%7Bts%2C%20js%7D&f=%2Fmyapp%2Freadme.md%2C%2Fmyapp%2Fconfig%2Fstaging.js%2C%2Fmyapp%2Fconfig%2Fproduction.js%2C%2Fmyapp%2Fsrc%2Fservices%2Futils.js%2C%2Fmyapp%2Fsrc%2Fservices%2Ftimezone.ts%2C%2Fmyapp%2Fsrc%2Fcontrollers%2Fhealth.js%2C%2Fmyapp%2Fsrc%2Fcontrollers%2Fuser.module.ts%2C%2Fmyapp%2Fassets%2Flogo.png%2C%2Fmyapp%2Fassets%2Flogo_small.png%2C%2Fmyapp%2Fassets%2Ffavicon.ico). You can try: `stylelint "**/*.{ts,tsx}" -f verbose`.

**Comment by jeffreyhshapiro** [comment]
FWIW, I'm experiencing the same thing but in v13.3.0 (also tried with v13.3.1 with the same result)

> Clearly describe the bug

I get the error `RangeError: Maximum call stack size exceeded` when running stylelint. It worked as expected before bumping versions.

> Which rule, if any, is the bug related to?

TBD

> What code is needed to reproduce the bug?

My code is conceptually similar to this setup-

It specifically looks like the code fails when `generateStyleRules()` is interpolated inside the `css` helper function.

```js
const generateStyleRules = () => `
  color: black;
`;

const StyledComponent = styled.div`
  margin: 8px 0;

  ${() =>
    css`
      div {
        ${generateStyleRules()};
      }
    `}
`;
```

> What stylelint configuration is needed to reproduce the bug?

> Which version of stylelint are you using?

`13.3.0`

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

CLI via npm script- `stylelint '**/*.js'`

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

I don't think so.

> What did you expect to happen?

It should pass linting

> What actually happened (e.g. what warnings or errors did you get)?

```shell
RangeError: Maximum call stack size exceeded
```

**Comment by Dangoo** [comment]
Can confirm it's fixed - thanks!
